x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -845,8 +845,9 @@ public Query prefixQuery(
 
         @Override
         public Query termsQuery(Collection<?> values, SearchExecutionContext context) {
+            HashSet<?> dedupe = new HashSet<>(values);
             BooleanQuery.Builder bq = new BooleanQuery.Builder();
-            for (Object value : values) {
+            for (Object value : dedupe) {
```

Comment:
I think this can over duplicate clauses. Are there other places where a similar change is needed to avoid duplicate wildcard clauses? I think this only covers queries targeting field of type wildcard?